### PR TITLE
feat(data): add simulations

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

Closes: #929 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/data/simulation/genesis.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package simulation
+
+import "github.com/cosmos/cosmos-sdk/types/module"
+
+// RandomizedGenState generates a random GenesisState for the data module.
+func RandomizedGenState(simState *module.SimulationState) {
+	// TODO: #186
+}
```
Comment (ryanchristo) [inline x/data/simulation/genesis.go:8]: Updated the description in #186 to make sure we include this update when we add genesis import and export.

**Review by ryanchristo — APPROVED** [review]
Looks good to me.

**Review by aaronc — APPROVED** [review]

File: x/data/simulation/operations.go [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+			sender.PrivKey,
+		)
+		if err != nil {
+			return simtypes.NoOpMsg(ModuleName, TypeMsgAnchor, "unable to generate mock tx"), nil, err
+		}
+
+		_, _, err = app.Deliver(txGen.TxEncoder(), tx)
+		if err != nil {
+			return simtypes.NoOpMsg(ModuleName, TypeMsgAnchor, "unable to deliver tx"), nil, err
+		}
```
Comment (robert-zaremba) [inline x/data/simulation/operations.go:146]: lines 127-141 are "almost" repeated in other functions below. Let's create a new helper function to generate and deliver a transaction

File: x/data/simulation/operations.go [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+		weightMsgAnchor           int
+		weightMsgAttest           int
+		weightMsgDefineResolver   int
+		weightMsgRegisterResolver int
+	)
+
+	appParams.GetOrGenerate(cdc, OpWeightMsgAnchor, &weightMsgAnchor, nil,
+		func(_ *rand.Rand) {
+			weightMsgAnchor = WeightAnchor
+		},
```
Comment (robert-zaremba) [inline x/data/simulation/operations.go:61]: it seams that this function is not needed (we could use a noop function). Why don't we set the params directly in lines 52-55?

**Pull request merged**